server/worker/apple_mdm.go

```diff
@@ -196,15 +314,25 @@ func QueueAppleMDMJob(
 	if teamID != nil {
 		attrs = append(attrs, "team_id", *teamID)
 	}
+	if len(enrollmentCommandUUIDs) > 0 {
+		attrs = append(attrs, "enrollment_commands", enrollmentCommandUUIDs)
+	}
 	level.Info(logger).Log(attrs...)
 
 	args := &appleMDMArgs{
-		Task:            task,
-		HostUUID:        hostUUID,
-		TeamID:          teamID,
-		EnrollReference: enrollReference,
+		Task:               task,
+		HostUUID:           hostUUID,
+		TeamID:             teamID,
+		EnrollReference:    enrollReference,
+		EnrollmentCommands: enrollmentCommandUUIDs,
+	}
+
+	// the release device task is always added with a delay
+	var delay time.Duration
+	if task == AppleMDMPostDEPReleaseDeviceTask {
+		delay = 30 * time.Second
```

Comment:
This task is always delayed, so that the device has the time to process the initial enrollment commands and a Reconcile profiles job has a chance to run.